scylla/tests/integration/session/schema_agreement.rs

```diff
@@ -0,0 +1,202 @@
+                .await
+                .unwrap();
+
+            let host_ids = calculate_proxy_host_ids(&proxy_uris, &translation_map, &session);
+
+            {
+                // Case 1: Paused node is a coordinator for DDL.
+                // DDL needs to fail.
+                let result = run_some_ddl_with_paused_node(
+                    NodeIdentifier::HostId(host_ids[1]),
+                    1,
+                    &session,
+                    &mut running_proxy,
+                )
+                .await;
+                assert_matches!(
+                    result,
+                    Err(ExecutionError::SchemaAgreementError(
+                        SchemaAgreementError::RequestError(
+                            RequestAttemptError::BrokenConnectionError(_)
+                        )
+                    ))
+                )
+            }
+
+            {
+                // Case 2: Paused node is NOT a coordinator for DDL.
+                // DDL should succeed, because auto schema agreement only needs available nodes to agree.
+                let result = run_some_ddl_with_paused_node(
+                    NodeIdentifier::HostId(host_ids[2]),
+                    1,
+                    &session,
+                    &mut running_proxy,
+                )
+                .await;
+                assert_matches!(result, Ok(_))
+            }
+
+            {
+                // Case 3: Paused node is a coordinator for DDL, and is used by control connection.
+                // It is the same as case 1, but paused node is also control connection.
+                // DDL needs to fail.
+                let result = run_some_ddl_with_paused_node(
+                    NodeIdentifier::HostId(host_ids[0]),
+                    0,
+                    &session,
+                    &mut running_proxy,
+                )
+                .await;
+                assert_matches!(
+                    result,
+                    Err(ExecutionError::SchemaAgreementError(
+                        SchemaAgreementError::RequestError(
+                            RequestAttemptError::BrokenConnectionError(_)
+                        )
+                    ))
+                )
+            }
+
+            {
+                // Case 4: Paused node is NOT a coordinator for DDL, but is used by control connection.
+                // It is the same as case 2, but paused node is also control connection.
+                // DDL should succeed, because auto schema agreement only needs available nodes to agree,
+                // and control connection is not used for that at all.
+                let result = run_some_ddl_with_paused_node(
+                    NodeIdentifier::HostId(host_ids[1]),
+                    0,
+                    &session,
+                    &mut running_proxy,
+                )
+                .await;
+                assert_matches!(result, Ok(_))
+            }
+
+            running_proxy
```

Comment:
:wrench: I'd also like to see a test case for the new `CoordinatorAbsent` (I may have changed the name) variant.

Reply:
I'm against any timing-based synchronization, it is a straight path towards flaky tests. Tomorrow I'll look into writing this test without using delays, but I'm not sure it will be possible with the current capabilities of the proxy.